pkg/acm/policy.go

```diff
@@ -0,0 +1,239 @@
+	}
+
+	glog.V(100).Infof("Creating the policy %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
+// Delete removes a policy from a cluster.
+func (builder *PolicyBuilder) Delete() (*PolicyBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting the policy %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		return builder, fmt.Errorf("policy cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.Delete(context.TODO(), builder.Definition)
+
+	if err != nil {
+		return builder, fmt.Errorf("can not delete policy: %w", err)
+	}
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// Update renovates the existing policy object with the policy definition in builder.
+func (builder *PolicyBuilder) Update(force bool) (*PolicyBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating the policy object", builder.Definition.Name)
+
+	err := builder.apiClient.Update(context.TODO(), builder.Definition)
+
+	if err != nil {
+		if force {
+			glog.V(100).Infof(
+				"Failed to update the policy object %s. "+
+					"Note: Force flag set, executed delete/create methods instead", builder.Definition.Name)
+
+			builder, err := builder.Delete()
+
+			if err != nil {
+				glog.V(100).Infof(
+					"Failed to update the policy object %s, "+
+						"due to error in delete function", builder.Definition.Name)
+
+				return nil, err
+			}
+
+			return builder.Create()
+		}
+	}
+
+	return builder, err
+}
+
+// validate will check that the builder and builder definition are properly initialized before
+// accessing any member fields.
+func (builder *PolicyBuilder) validate() (bool, error) {
+	resourceCRD := "policy"
+
+	if builder == nil {
+		glog.V(100).Infof("The %s builder is uninitialized", resourceCRD)
+
+		return false, fmt.Errorf("error: received nil %s builder", resourceCRD)
+	}
+
+	if builder.Definition == nil {
+		glog.V(100).Infof("The %s is undefined", resourceCRD)
+
+		builder.errorMsg = msg.UndefinedCrdObjectErrString(resourceCRD)
+	}
+
+	if builder.apiClient == nil {
+		glog.V(100).Infof("The %s builder apiclient is nil", resourceCRD)
+
+		builder.errorMsg = fmt.Sprintf("%s builder cannot have nil apiClient", resourceCRD)
+	}
+
+	if builder.errorMsg != "" {
+		glog.V(100).Infof("The %s builder has error message: %s", resourceCRD, builder.errorMsg)
+
+		return false, fmt.Errorf(builder.errorMsg)
+	}
+
+	return true, nil
+}
+
+// ListPoliciesInAllNamespaces returns a cluster-wide pod inventory.
+func ListPoliciesInAllNamespaces(apiClient *clients.Settings, options metaV1.ListOptions) ([]*PolicyBuilder, error) {
```

Comment:
I think the List should be part of a different file ( policylist.go ) that will handle lists of policies .

There are few examples in different packages.